analyzer/src/main/kotlin/managers/utils/NuGetSupport.kt

```diff
@@ -43,296 +56,278 @@ import org.ossreviewtoolkit.model.Scope
-            sortedSetOf<String>().apply {
-                val license = node["license"]?.textValue() ?: node["licenseUrl"].textValueOrEmpty()
-                // Most nuget packages only provide a "licenseUrl" which counts as a declared license.
-                if (license.isNotEmpty()) {
-                    add(license)
-                }
-            }
+    // See https://docs.microsoft.com/en-us/nuget/api/catalog-resource#package-details-catalog-items.
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    data class PackageDetails(
+        val id: String,
+        val version: String,
+        val packageHash: String,
+        val packageHashAlgorithm: String,
+        val packageSize: Int,
+
+        val description: String? = null,
+        val projectUrl: String? = null,
+        val dependencyGroups: List<DependencyGroup> = emptyList(),
+    )
 
-        private fun extractRemoteArtifact(node: JsonNode, nupkgUrl: String): RemoteArtifact {
-            val encodedHash = node["packageHash"].textValueOrEmpty()
-            val hashAlgorithm = node["packageHashAlgorithm"].textValueOrEmpty()
-            val sri = hashAlgorithm.replace("-", "") + "-" + encodedHash
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    data class DependencyGroup(
+        val targetFramework: String? = null,
+        val dependencies: List<Dependency> = emptyList()
+    )
 
-            return RemoteArtifact(nupkgUrl, Hash.create(sri))
-        }
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    data class Dependency(
+        val id: String,
+        val range: String
+    )
 
-        private fun getCatalogURL(registrationNode: JsonNode): String =
-            registrationNode["catalogEntry"].textValueOrEmpty()
+    // See https://docs.microsoft.com/en-us/nuget/reference/nuspec.
+    data class PackageSpec(
+        val metadata: MetaData
+    )
 
-        private fun getNuspecURL(registrationNode: JsonNode): String =
-            registrationNode["packageContent"].textValueOrEmpty()
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    data class MetaData(
+        // See https://docs.microsoft.com/en-us/nuget/reference/nuspec#required-metadata-elements
+        val id: String,
+        val version: String,
+        val description: String,
+        val authors: String,
+
+        // See https://docs.microsoft.com/en-us/nuget/reference/nuspec#optional-metadata-elements.
+        val licenseUrl: String? = null,
+        val license: License? = null,
+        val repository: Repository? = null,
+    )
 
-        private fun extractVersion(range: String): String {
-            if (range.isEmpty()) return ""
-            val rangeReplaces = range.replace("[", "")
-                .replace(" ", "")
-                .replace(")", "")
-            return rangeReplaces.split(",").elementAt(0)
-        }
-    }
+    data class License(
+        @JacksonXmlProperty(isAttribute = true)
+        val type: String,
+        @JacksonXmlProperty(isAttribute = true)
```

Comment:
In line 81 there is an empty line while in line 141 there isn't.
Could be made consistent. Similar multiple times below.

Reply:
I've removed the blank line in 81.